src/Data/Conduit/BZlib.hs

```diff
@@ -172,13 +176,39 @@ decompress DecompressParams {..} = do
         yield $ fromJust mbout
       availIn <- liftIO $ peek $ p'bz_stream'avail_in ptr
       if availIn > 0
-        then yields ptr
+        then
+            -- bzip2 files can contain multiple concatenated streams, but the
+            -- API requires that we close the stream and start a new
+            -- decompression session.
+            if ret == c'BZ_STREAM_END
+                then do
+                    dataIn <- liftIO $ peek $ p'bz_stream'next_in ptr
+                    unread <- liftIO $ S.packCStringLen (dataIn, fromEnum availIn)
+                    leftover unread
+                    return False
+                else yields ptr
         else return $ ret == c'BZ_OK
 
+-- Decompress all the compressed bzip2 streams in the input, as the bzip2
+-- command line tool.
+decompress
+  :: MonadResource m
+     => DecompressParams -- ^ Decompress parameter
+     -> ConduitT S.ByteString S.ByteString m ()
+decompress params = do
+    decompress1 params
+    next <- await
+    case next of
+        Nothing -> return ()
+        Just bs -> do

```

Comment:
Depending on data source, it's theoretically possible that at end of stream we may receive an empty ByteString. This code would treat such a chunk as the start of a new stream. It may be better to integrate this logic more closely with the multi stream detection above.

Reply:
That's a fair catch. I'll change the logic to ignore empty `ByteString`s